.editorconfig

```diff
@@ -0,0 +1,11 @@
+# top-most EditorConfig file
```

Comment:
As this adds a new top-level config file, can you say more about what it does and what (if anything, such as a plug-in) is needed for editors to adhere to it?

Reply:
This sets consistent whitespace characters and line endings for various editors. We use it in Vega-Lite but I'm happy to remove it from this PR. 

Some editors (e.g. github, vs) don't need any additional plugins. There are plugins for various editors at https://editorconfig.org/#download.

Reply:
I'm all for having a useful config and this looks totally reasonable. I just wanted to understand it before merging. I'm also open to using prettier in a consistent way across the Vega modules (though obviously that would be the subject of a different PR).